packages/core/src/workflow/aggregator.js

```diff
@@ -0,0 +1,83 @@
+import { Aggregate, Index } from '@web3-storage/data-segment'
+
+/**
+ * @param {object} props
+ * @param {import('../types.js').PieceQueue} props.pieceQueue
+ * @param {import('../types.js').AggregateQueue} props.aggregateQueue
+ * @param {import('@web3-storage/data-segment').PaddedPieceSize} [props.builderSize]
+ * @returns {import('../types.js').AggregatorWorkflowResponse}
+ */
+export async function build ({ pieceQueue, aggregateQueue, builderSize }) {
+  // Get maximum number of pieces that can make it to an aggregate
+  const size = builderSize || Aggregate.DEFAULT_DEAL_SIZE
+  const limit = Index.maxIndexEntriesInDeal(size)
+
+  // Get piece list and immediately leave if no pieces
+  const pieceListResponse = await pieceQueue.peek({
+    // TODO: We need to see how this behaves with large items
+    // Likely paginates...
+    limit
+  })
+  if (pieceListResponse.error) {
+    return { error: pieceListResponse.error }
+  } else if (!pieceListResponse.ok.length) {
+    return {
+      ok: {
+        count: 0
+      }
+    }
+  }
+
+  // Attempt to create aggregate
+  const aggregate = buildAggregate(pieceListResponse.ok, {
+    size
+  })
+
+  // Write to aggregate queue
+  const { error } = await aggregateQueue.put(aggregate)
+  if (error) {
+    return { error }
+  }
+
+  return {
+    ok: {
+      count: 1
+    }
+  }  
+}
+
+/**
+ * @param {import('../types.js').Inserted<import('../types.js').Inclusion>[]} unsortedPieces
+ * @param {object} [options]
+ * @param {import('@web3-storage/data-segment').PaddedPieceSize} [options.size]
+ */
+function buildAggregate (unsortedPieces, options = {}) {
+  const size = options.size
+  const builder = Aggregate.createBuilder({
+    size
+  })
+  // Sort pieces by size as naive implementation of an aggregate
+  const pieces = [
+    ...unsortedPieces
+  ].sort((a, b) => Number(a.size - b.size))
+  const addedPieces = []
+
+  for (const item of pieces) {
+    try {
+      builder.write({
+        root: item.piece.multihash.digest,
+        size: item.size
+      })
+      addedPieces.push(item.piece)
+    } catch {}
+  }
+
+  // TODO: Figure out https://github.com/web3-storage/data-segment/pull/10#discussion_r1261081931
+  const aggregate = builder.build()
```

Comment:
@Gozala here was what I meant on that comment in `data-segment` PR. We define an aggregate size, but when we can really build any size. We should be able to better track what we used (and/or perhaps be able to give a minimumSize that would make `build()` fail if there is not enough

Reply:
@vasco-santos I have created an issue https://github.com/web3-storage/data-segment/issues/15 with all the details. I won't repeat them here, but it should be already possible to tell how much size has been utilized already, so you could decide here whether it's good to build or not. We could expose them as getters also to make it even nicer to use.

It will not through, but it will give you a number that you can compare to `minimumSize` here so you could decide.